pkgs/development/interpreters/python/mk-python-derivation.nix

```
@@ -115,343 +120,343 @@ let
+      doCheck = false;
+      doInstallCheck = prev.doCheck or true;
+      nativeInstallCheckInputs = nativeCheckInputs;
+      installCheckInputs = checkInputs;
+
+      postFixup =
+        optionalString (!dontWrapPythonPrograms) ''
+          wrapPythonPrograms
+        ''
+        + prev.postFixup or "";
+
+      # Python packages built through cross-compilation are always for the host platform.
+      disallowedReferences = optionals (python.stdenv.hostPlatform != python.stdenv.buildPlatform) [
+        python.pythonOnBuildForHost
+      ];
+
+      outputs = outputs ++ optional withDistOutput "dist";
+
+      passthru =
+        {
+          inherit disabled;
         }
-        // optionalAttrs (attrs ? disabledTests) {
-          # `escapeShellArgs` should be used as well as `disabledTestPaths`,
-          # but some packages rely on existing raw strings.
-          disabledTests = attrs.disabledTests;
+        // prev.passthru or { }
+        // {
+          updateScript =
+            let
+              filename = head (splitString ":" final.meta.position);
+            in
+            prev.passthru.updateScript or [
+              update-python-libraries
+              filename
+            ];
         }
-        // optionalAttrs (attrs ? pytestFlagsArray) {
-          pytestFlagsArray = attrs.pytestFlagsArray;
+        // optionalAttrs (dependencies != [ ]) {
+          inherit dependencies;
         }
-        // optionalAttrs (attrs ? unittestFlagsArray) {
-          unittestFlagsArray = attrs.unittestFlagsArray;
+        // optionalAttrs (optional-dependencies != { }) {
+          inherit optional-dependencies;
         }
-      )
-    )
-  );
+        // optionalAttrs (build-system != [ ]) {
+          inherit build-system;
+        };
+
+      meta = {
+        # default to python's platforms
+        platforms = python.meta.platforms;
+        isBuildPythonPackage = python.meta.platforms;
+      } // meta;
+    }
+    // optionalAttrs (prev ? checkPhase) {
+      # If given use the specified checkPhase, otherwise use the setup hook.
+      # Longer-term we should get rid of `checkPhase` and use `installCheckPhase`.
+      installCheckPhase = prev.checkPhase;
+    }
+    // optionalAttrs (prev.doCheck or true) (
+      optionalAttrs (disabledTestPaths != [ ]) {
+        disabledTestPaths = escapeShellArgs disabledTestPaths;
+      }
+      // optionalAttrs (prev ? disabledTests) {
+        # `escapeShellArgs` should be used as well as `disabledTestPaths`,
+        # but some packages rely on existing raw strings.
+        disabledTests = prev.disabledTests;
+      }
+      // optionalAttrs (prev ? pytestFlagsArray) {
+        pytestFlagsArray = prev.pytestFlagsArray;
+      }
+      // optionalAttrs (prev ? unittestFlagsArray) {
+        unittestFlagsArray = prev.unittestFlagsArray;
+      }
+    );
 
 in
+origAttrs:
+let
+  unfixed = (lib.toFunction origAttrs) { };
+  stdenv = unfixed.stdenv or python.stdenv;
```

```suggestion
  stdenv = origAttrs.stdenv or python.stdenv;
```

if `origAttrs` is `{ stdenv = ...; }`, `unfixed` will be a callable equivalent to `finalAttrs: { stdenv = "..."; }`. When `origAttrs` is a function, it is impossible to specify `stdenv` as its arguments and get it out before fixing the arguments.

`buildPythonPackage.override { stdenv = ...; }` is the only possible and reasonable way to specify custom `stdenv`. That's what #271762 is for.